dbms/src/Functions/FunctionsString.cpp

```diff
@@ -5022,52 +5027,410 @@ class FunctionPosition : public IFunction
+            }
+            else
+            {
+                constVector(c0_const->getValue<String>(), c1_string->getChars(), c1_string->getOffsets(), vec_res);
+            }
+        }
+        else if (c0_string && c1_string)
+        {
+            vectorVector(
+                c0_string->getChars(),
+                c0_string->getOffsets(),
+                c1_string->getChars(),
+                c1_string->getOffsets(),
+                vec_res);
+        }
+        else if (c0_string && c1_const)
+        {
+            vectorConst(c0_string->getChars(), c0_string->getOffsets(), c1_const->getValue<String>(), vec_res);
+        }
+        else
+            throw Exception(
+                "Illegal columns " + block.getByPosition(arguments[0]).column->getName() + " and "
+                    + block.getByPosition(arguments[1]).column->getName() + " of arguments of function " + getName(),
+                ErrorCodes::ILLEGAL_COLUMN);
+
+        block.getByPosition(result).column = std::move(col_res);
+    }
+
+    void vectorVector(
+        const ColumnString::Chars_t & col0_data,
+        const ColumnString::Offsets & col0_offsets,
+        const ColumnString::Chars_t & col1_data,
+        const ColumnString::Offsets & col1_offsets,
+        PaddedPODArray<Int64> & res) const
+    {
+        size_t row_num = col0_offsets.size();
+        ColumnString::Offset prev_col0_str_offset = 0;
+        ColumnString::Offset prev_col1_str_offset = 0;
+
+        String col0_container;
+        String col1_container;
+
+        std::vector<size_t> lens;
+
+        for (size_t i = 0; i < row_num; i++)
+        {
+            size_t col0_str_len = col0_offsets[i] - prev_col0_str_offset - 1;
+            size_t col1_str_len = col1_offsets[i] - prev_col1_str_offset - 1;
+
+            if unlikely (col0_str_len == 0)
+            {
+                res[i] = 1;
+            }
+            else
+            {
+                const StringRef & col0_collation_str = collator->sortKeyNoTrim(
+                    reinterpret_cast<const char *>(&col0_data[prev_col0_str_offset]),
+                    col0_str_len,
+                    col0_container);
+                const StringRef & col1_collation_str = collator->convert(
+                    reinterpret_cast<const char *>(&col1_data[prev_col1_str_offset]),
+                    col1_str_len,
+                    col1_container,
+                    &lens);
+                void * res_start = memmem(
+                    col1_collation_str.data,
+                    col1_collation_str.size,
+                    col0_collation_str.data,
+                    col0_collation_str.size);
+
+                if (res_start == nullptr)
+                    res[i] = 0;
+                else
+                {
+                    size_t pos = reinterpret_cast<const char *>(res_start) - col1_collation_str.data;
+                    res[i] = 1 + getPositionWithCollationString(pos, lens);
+                }
+            }
```

Comment:
May extract L5307 to L5319 as a common function inside the class.

Reply:
> May extract L5307 to L5319 as a common function inside the class.

done